Analytics-CSharp/Segment/Analytics/Compat/Migration.cs

```diff
@@ -0,0 +1,81 @@
+using System.Collections.Generic;
+using Segment.Analytics;
+using Segment.Serialization;
+
+namespace AnalyticsNetMigrationHelper {
+    public class Traits : Dictionary<string, object>
+    {
+    }
+
+    public class Properties  : Dictionary<string, object>
+    {
+    }
+
+    public static class AnalyticsExtensions
+    {
+        public static void Track(this Analytics analytics, string userId, string eventName)
+        {
+            analytics.Track(eventName, new JsonObject() { {"userId", userId} });
+        }
+        public static void Track(this Analytics analytics, string userId, string eventName, IDictionary<string, object> traits)
+        {
+            traits.Add("userId", userId);
+            analytics.Track(eventName, JsonUtility.FromJson<Segment.Serialization.JsonObject>(JsonUtility.ToJson(traits)));
+        }
+
+        public static void Screen(this Analytics analytics, string userId, string eventName)
+        {
+            analytics.Screen(eventName, new JsonObject() { {"userId", userId} });
+        }
+        public static void Screen(this Analytics analytics, string userId, string eventName, IDictionary<string, object> traits)
+        {
+            traits.Add("userId", userId);
+            analytics.Screen(eventName, JsonUtility.FromJson<Segment.Serialization.JsonObject>(JsonUtility.ToJson(traits)));
+        }
+
+        public static void Page(this Analytics analytics, string userId, string eventName)
+        {
+            analytics.Page(eventName, new JsonObject() { {"userId", userId} });
+        }
+        public static void Page(this Analytics analytics, string userId, string eventName, IDictionary<string, object> traits)
+        {
+            traits.Add("userId", userId);
+            analytics.Page(eventName, JsonUtility.FromJson<Segment.Serialization.JsonObject>(JsonUtility.ToJson(traits)));
+        }
+    }   
+
+    class UserIdPlugin : EventPlugin 
+    {
+        public override PluginType Type => PluginType.Enrichment;
+
+        public override TrackEvent Track(TrackEvent trackEvent)
+        {
+            if (trackEvent.Properties.ContainsKey("userId"))
```

Comment:
the body of this method can be factored out to a private method and be used across the other methods, less redundant code. we probably need to do some for group and alias, since in old sdk, these two methods also take `userId` as parameter. but we can handle them in the same way, since `Properties` and `Traits` are both JsonObject

Reply:
I was thinking something like the following. not necessary better, just less redundant code
```

class UserIdPlugin : EventPlugin
{
    public override PluginType Type { get; }

    public override RawEvent Execute(RawEvent incomingEvent)
    {
        switch (incomingEvent)
        {
            case TrackEvent e:
                PatchUserId(e, e.Properties);
                break;
            case PageEvent e:
                PatchUserId(e, e.Properties);
                break;
            case ScreenEvent e:
                PatchUserId(e, e.Properties);
                break;
            case GroupEvent e:
                PatchUserId(e, e.Traits);
                break;
        }

        return incomingEvent;
    }

    private void PatchUserId(RawEvent @event, JsonObject jsonObject)
    {
        if (jsonObject.ContainsKey("userId"))
        {
            @event.UserId = jsonObject.GetString("user_id");
            jsonObject.Remove("userId");
        }
    }
}
```